test/components/legendTests.ts

```diff
@@ -350,3 +350,183 @@ describe("Legend", () => {
     svg.remove();
   });
 });
+
+describe("InterpolatedColorLegend", () => {
+  var svg: D3.Selection;
+  var colorScale: Plottable.Scale.InterpolatedColor;
+
+  beforeEach(() => {
+    svg = generateSVG(400, 400);
+    colorScale = new Plottable.Scale.InterpolatedColor();
+  });
+
+  function assertBasicRendering(legend: Plottable.Component.InterpolatedColorLegend) {
+    var scaleDomain = colorScale.domain();
+    var legendElement: D3.Selection = (<any> legend)._element;
+
+    var swatches = legendElement.selectAll(".swatch");
```

Comment:
Sounds sort of like the API point on the `Plot`s. Should this be something more general?